### PR TITLE
File Slink

### PR DESCRIPTION
**Pull request by raayan**
Basic implementation of `File Metadata Service` which handles communication and conflict resolution between `server` and `core`. Basic iOS view of file metadata with hooks to `core` library.

### CODE REVIEW

File: core/Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "lockbook_core"
 version = "2.1.2"
-authors = ["Parth <parth@mehrotra.me>"]
+authors = ["Parth <parth@mehrotra.me>", "Raayan <raayan@raayanpillai.com"]
```
Comment (tvanderstad) [inline core/Cargo.toml:4]: missing '>'

File: core/Cargo.toml [hunk 2]
```diff
@@ -17,4 +17,4 @@ openssl = { version = "0.10", features = ["vendored"] }
 base64 = "0.11.0"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0.44"
-uuid = { version = "0.8", features = ["v4"] }
+uuid = { version = "0.8.1", features = ["v4"] }
```
Comment (tvanderstad) [inline core/Cargo.toml:20]: add a newline

File: core/src/db_provider.rs [hunk 3]
```diff
@@ -28,20 +28,31 @@ pub struct RamBackedDB<Schema: SchemaApplier> {
...
-
-        Ok(db)
+        match Schema::apply_schema(&db) {
+            Ok(_) => {
+                println!("Schema applied succesfully!");
+                Ok(db)
+            }
+            Err(err) => {
+                println!(
+                    "Table creation failure, probably already exists, continuing! Error: {:?}",
```
Comment (tvanderstad) [inline core/src/db_provider.rs:45]: is there a way you can check?

File: core/src/file_metadata.rs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+use serde::Serialize;
+
+#[derive(PartialEq, Debug, Serialize)]
+pub struct FileMetadata {
```
Comment (tvanderstad) [inline core/src/file_metadata.rs:4]: maybe duplicate of a struct in core/src/lockbook_api/get_updates.rs

File: ios_client/ios_client/Apis/CoreApi.swift [hunk 5]
```diff
@@ -0,0 +1,89 @@
+//
+//  LockbookApi.swift
+//  ios_client
+//
+//  Created by Raayan Pillai on 4/11/20.
+//  Copyright © 2020 Lockbook. All rights reserved.
```
Comment (tvanderstad) [inline ios_client/ios_client/Apis/CoreApi.swift:6]: are all rights actually reserved tho?

File: ios_client/ios_client/SceneDelegate.swift [hunk 6]
```diff
@@ -12,26 +12,33 @@ import SwiftUI
...
     func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
         // Use this method to optionally configure and attach the UIWindow `window` to the provided UIWindowScene `scene`.
         // If using a storyboard, the `window` property will automatically be initialized and attached to the scene.
         // This delegate does not imply the connecting scene or session are new (see `application:configurationForConnectingSceneSession` instead).
         
-        // Create the SwiftUI view that provides the window contents.
+        // Create the Lockbook Core Api with the path all our business happens
+        print(documentsDirectory)
+        let lockbookApi = CoreApi(documentsDirectory: documentsDirectory)
+//        let lockbookApi = FakeApi()
```
Comment (tvanderstad) [inline ios_client/ios_client/SceneDelegate.swift:29]: remove?

File: core/src/file_metadata_repo.rs [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+        FileRowMissing(NoneError)
+    }
+}
+
+pub trait FileMetadataRepo {
+    fn insert(db: &Connection, file_metadata: &FileMetadata) -> Result<(), Error>;
+    fn update(db: &Connection, file_metadata: &FileMetadata) -> Result<(), Error>;
+    fn get(db: &Connection, id: &String) -> Result<FileMetadata, Error>;
+    fn last_updated(db: &Connection) -> Result<i64, Error>;
+    fn dump(db: &Connection) -> Result<Vec<FileMetadata>, Error>;
```
Comment (Parth) [inline core/src/file_metadata_repo.rs:21]: ~Dump~

File: core/src/file_metadata_service.rs [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+        )
+        .map(|metadatas| {
+            metadatas
+                .into_iter()
+                .map(|t| FileMetadata {
+                    id: t.file_id,
+                    name: t.file_name,
+                    path: t.file_path,
+                    updated_at: t.file_metadata_version as i64,
+                    status: "Remote".to_string(),
```
Comment (Parth) [inline core/src/file_metadata_service.rs:59]: TODO here, could result in loss of local changes

File: core/src/file_metadata_service.rs [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+                })
+                .collect::<Vec<FileMetadata>>()
+        });
+
+        updates?
```
Comment (Parth) [inline core/src/file_metadata_service.rs:64]: `?` at callsite

**Pull request merged**